macros/union_data.sql

```diff
@@ -22,67 +22,88 @@
     default_variable,
     union_schema_variable,
     union_database_variable
-    ) %}
+    ) -%}
 
-{% if var(union_schema_variable, none) %}
+{%- if var(union_schema_variable, none) -%}
 
-    {% set relations = [] %}
+    {%- set relations = [] -%}
     
-    {% if var(union_schema_variable) is string %}
-    {% set trimmed = var(union_schema_variable)|trim('[')|trim(']') %}
-    {% set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") %}
-    {% else %}
-    {% set schemas = var(union_schema_variable) %}
-    {% endif %}
-
-    {% for schema in var(union_schema_variable) %}
-    {% set relation=adapter.get_relation(
+    {%- if var(union_schema_variable) is string -%}
+    {%- set trimmed = var(union_schema_variable)|trim('[')|trim(']') -%}
+    {%- set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") -%}
+    {%- else -%}
+    {%- set schemas = var(union_schema_variable) -%}
+    {%- endif -%}
+
+    {%- for schema in var(union_schema_variable) -%}
+    {%- set relation=adapter.get_relation(
         database=source(schema, table_identifier).database if var('has_defined_sources', false) else var(database_variable, default_database),
         schema=source(schema, table_identifier).schema if var('has_defined_sources', false) else schema,
         identifier=source(schema, table_identifier).identifier if var('has_defined_sources', false) else table_identifier
     ) -%}
     
-    {% set relation_exists=relation is not none %}
+    {%- set relation_exists=relation is not none -%}
 
-    {% if relation_exists %}
+    {%- if relation_exists -%}
+        {%- do relations.append(relation) -%}
+    {%- endif -%}
 
-    {% do relations.append(relation) %}
+    {%- endfor -%}
     
-    {% endif %}
-
-    {% endfor %}
-
-    {{ dbt_utils.union_relations(relations) }}
-
-{% elif var(union_database_variable, none) %}
+    {%- if relations != [] -%}
+        {{ dbt_utils.union_relations(relations) }}
+    {%- else -%}
+    {% if execute and not var('remove_empty_table_warnings', false) -%}{{ exceptions.warn("\n\nWarning: The " ~ table_identifier ~ " table was not found in your destination. The package will create a completely empty staging model as to not break downstream transformations.\n") }}{% endif -%}
```

Comment:
Now that I think about this, is "Warning" the right word here? A warning to me indicates something needs to be looked into, this may not 🤷. Is there a less severe word we could possibly use here?

Reply:
Additionally, what are your thoughts in providing the variable to disable in the message itself? Just so customers are directly aware of it and can turn it off without needing to inspect our docs?

Reply:
>Now that I think about this, is "Warning" the right word here? A warning to me indicates something needs to be looked into, this may not 🤷. Is there a less severe word we could possibly use here?

Yeah honestly i thought about "FYI" or "Heads up" lol

And I really like that idea!!